airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/datetime_stream_slicer.py

```diff
@@ -145,28 +145,19 @@ def stream_slices(self, sync_mode: SyncMode, stream_state: Mapping[str, Any]) ->
         kwargs = {"stream_state": stream_state}
         end_datetime = min(self.end_datetime.get_datetime(self.config, **kwargs), datetime.datetime.now(tz=self._timezone))
         lookback_delta = self._parse_timedelta(self.lookback_window.eval(self.config, **kwargs) if self.lookback_window else "0d")
-        start_datetime = self.start_datetime.get_datetime(self.config, **kwargs) - lookback_delta
-        start_datetime = min(start_datetime, end_datetime)
-        if self.cursor_field.eval(self.config, stream_state=stream_state) in stream_state:
-            cursor_datetime = self.parse_date(stream_state[self.cursor_field.eval(self.config)])
-        else:
-            cursor_datetime = start_datetime
-
-        start_datetime = max(cursor_datetime, start_datetime)
 
-        state_cursor_value = stream_state.get(self.cursor_field.eval(self.config, stream_state=stream_state))
+        earliest_possible_start_datetime = min(self.start_datetime.get_datetime(self.config, **kwargs), end_datetime)
+        cursor_datetime = self._calculate_cursor_datetime_from_state(stream_state)
+        start_datetime = max(earliest_possible_start_datetime, cursor_datetime) - lookback_delta
 
-        if state_cursor_value:
-            state_date = self.parse_date(state_cursor_value)
-        else:
-            state_date = None
-        if state_date:
-            # If the input_state's date is greater than start_datetime, the start of the time window is the state's next day
-            next_date = state_date + datetime.timedelta(days=1)
-            start_datetime = max(start_datetime, next_date)
         dates = self._partition_daterange(start_datetime, end_datetime, self._step)
         return dates
 
+    def _calculate_cursor_datetime_from_state(self, stream_state: Mapping[str, Any]) -> datetime.datetime:
+        if self.cursor_field.eval(self.config, stream_state=stream_state) in stream_state:
+            return self.parse_date(stream_state[self.cursor_field.eval(self.config)]) + datetime.timedelta(days=1)
```

Comment:
Why do we have this `+ datetime.timedelta(days=1)`here? What does it mean? This isn't so clear to me and I would like to extract this as a constant with a clear name as to what it does. I see [a comment](https://github.com/airbytehq/airbyte/pull/20156/files#diff-3a6ff2f6eca71b4e166b1c8c852becb143ad4e079f5308967ff52828e747f035L164) here describing what we do, but I'm not sure as to why we do it

Reply:
Just for extra context about incremental syncs a little bit. For streams that are configured as incrementals. The first time we sync, we will sync all the data up to the current day. Then at the end we save a stream state checkpoint to remind us where we stopped. On the second iteration, we'll use that `stream_state` checkpoint to tell us to begin at that point.

So idea here is that if we have received a `stream_state` as a datetime input, that would indicate that we had successfully completed an incremental sync before the current sync we're processing. And now on this current one, we want to start our next sync on the next day because we had already synced the current day in the prior sync.

For ex:
If we sync up to 12/7. We persist 12/7 into the state database. So that way on 12/8 when we perform our next sync, we only start getting records beginning on 12/8. This is pretty common for our connectors where the cursor date like created is `12-07-2022`. But that does raise a good question about how accurate this for integrations whose granularity is at the hours, minutes or even seconds, we could be missing thing. I think we should follow up on this. But it might fall a little outside the scope of this fix since this was how our existing behavior was.